packages/orbit/src/utils.ts

```diff
@@ -0,0 +1,16 @@
+import { type FilterOperator } from '@rabbitholegg/questdk'
+
+type Amount = FilterOperator | BigInt | number | string | undefined
+
+export function getUnlockTime(amount: Amount): FilterOperator | undefined {
+  if (amount === undefined) return undefined
+
+  // in seconds
+  const now = Math.floor(new Date().getTime() / 1000)
+
+  if (typeof amount === 'object') {
+    const [operator, value] = Object.entries(amount)[0]
+    return { [operator]: BigInt(value) + BigInt(now) }
+  }
+  return { $gte: BigInt(amount) + BigInt(now) }
```

Comment:
Can you talk me through this? It looks like we're basically requiring that the unlock time is 1 second in the future for every 1 wei of value?

Reply:
Nevermind - I understand now.

I would change `amount: Amount` to `duration: Amount` or even `duration: Duration`. Also I think we should have that type somewhere but if we don't let's export it from utils vs define it here. That's a pretty useful/re-usable one.

Reply:
duration will be passed in as the amount of time in seconds.

The `_unlock_time` is a timestamp which denotes the exact time that the stake will become unlocked.

The logic here is that The current time (date.now()) + the duration in seconds should meet or exceed the `_unlock_time` timestamp shown in the transaction to pass (if the operator is $gte)